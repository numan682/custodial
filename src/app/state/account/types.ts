import { Transaction } from 'app/state/transaction/types'
import { ErrorPayload } from 'types/errors'
import { StringifiedBigInt } from 'types/StringifiedBigInt'

export interface BalanceDetails {
  available: StringifiedBigInt | null
  /** This is delayed in getAccount by 20 seconds on oasisscan and 5 seconds on oasismonitor. */
  debonding: StringifiedBigInt | null
  /** This is delayed in getAccount by 20 seconds on oasisscan and 5 seconds on oasismonitor. */
  delegations: StringifiedBigInt | null
  /** This is delayed in getAccount by 20 seconds on oasisscan and 5 seconds on oasismonitor. */
  total: StringifiedBigInt | null
}

export interface Allowance {
  address: string
  amount: StringifiedBigInt
}

export interface Account extends BalanceDetails {
  address: string
  allowances?: Allowance[]
}

/* --- STATE --- */
export interface AccountState extends Account {
  loading: boolean
  accountError?: ErrorPayload
  transactions: Transaction[]
  transactionsError?: ErrorPayload
}
