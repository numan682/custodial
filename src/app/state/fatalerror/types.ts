/* --- STATE --- */
export interface FatalErrorPayload {
  message: string
  stack?: string
  sagaStack?: string
}

export interface FatalErrorState {
  error?: FatalErrorPayload
}
