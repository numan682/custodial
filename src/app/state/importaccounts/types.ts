import { ErrorPayload } from 'types/errors'
import { WalletType } from '../wallet/types'
import { BalanceDetails } from '../account/types'

/* --- STATE --- */
export interface ImportAccountsListAccount {
  address: string
  balance?: BalanceDetails
  path: number[]
  pathDisplay: string
  privateKey?: string
  publicKey: string
  selected: boolean
  type: WalletType
}

export enum ImportAccountsStep {
  Idle = 'idle',
  OpeningUSB = 'opening_usb',
  LoadingAccounts = 'loading_accounts',
  LoadingBalances = 'loading_balances',
}

export interface ImportAccountsState {
  accounts: ImportAccountsListAccount[]
  error?: ErrorPayload
  showAccountsSelectionModal: boolean
  accountsSelectionPageNumber: number
  step: ImportAccountsStep
}
