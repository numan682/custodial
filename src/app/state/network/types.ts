/* --- STATE --- */

export type NetworkType = 'local' | 'testnet' | 'mainnet'

export interface NetworkState {
  selectedNetwork: NetworkType

  /** Ticker symbol (ROSE, TEST, etc.) */
  ticker: string

  /** chainContext / Genesis Hash */
  chainContext: string

  /** Current epoch */
  epoch: number

  /** Minimum staking amount */
  minimumStakingAmount: number
}
