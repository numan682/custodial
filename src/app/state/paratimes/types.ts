import { StringifiedBigInt } from 'types/StringifiedBigInt'
import { ErrorPayload } from 'types/errors'
import { ParaTime } from '../../../config'

export enum TransactionFormSteps {
  TransferType = 'transferType',
  ParaTimeSelection = 'paraTimeSelection',
  TransactionRecipient = 'transactionRecipient',
  TransactionAmount = 'transactionAmount',
  TransactionConfirmation = 'transactionConfirmation',
  TransactionSummary = 'transactionSummary',
  TransactionError = 'transactionError',
}

export enum TransactionTypes {
  Deposit = 'deposit',
  Withdraw = 'withdraw',
}

export interface TransactionForm {
  amount: string
  confirmTransfer: boolean
  confirmTransferToValidator: boolean
  confirmTransferToForeignAccount: boolean
  defaultFeeAmount: string
  ethPrivateKey: string
  feeAmount: string
  feeGas: string
  paraTime?: ParaTime
  recipient: string
  type: TransactionTypes | undefined
}

export interface ParaTimesState {
  balance: StringifiedBigInt
  isLoading: boolean
  transactionError?: ErrorPayload
  transactionForm: TransactionForm
  transactionFormStep: TransactionFormSteps
}

export type Runtime = {
  address: string
  id: string
  decimals: number
  feeGas: bigint
}

export type ParaTimeTransaction = Pick<
  TransactionForm,
  'amount' | 'ethPrivateKey' | 'feeAmount' | 'feeGas' | 'recipient' | 'type'
>
