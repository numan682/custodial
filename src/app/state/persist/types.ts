import { RootState } from 'types'
// eslint-disable-next-line @typescript-eslint/ban-types
interface PreserveAliasName extends String {}

export interface KeyWithSalt {
  key: Uint8Array
  salt: Uint8Array
}
/** Redux can't serialize Uint8Arrays so we base64 encode and stringify */
export type StringifiedKeyWithSalt = string & PreserveAliasName

export interface EncryptedObject {
  secretbox: Uint8Array
  nonce: Uint8Array
  salt: Uint8Array
}
/** Redux can't serialize Uint8Arrays so we base64 encode and stringify */
export type EncryptedString = string & PreserveAliasName
// Hover to check if inferred variable type is EncryptedString (not string)
export const testPreserveAliasName = '{}' as EncryptedString

export interface SetUnlockedRootStatePayload {
  persistedRootState: PersistedRootState
  stringifiedEncryptionKey: StringifiedKeyWithSalt
}

/* --- STATE --- */
export interface PersistState {
  hasPersistedProfiles: boolean
  isPersistenceUnsupported: boolean
  loading: boolean
  stringifiedEncryptionKey: undefined | 'skipped' | StringifiedKeyWithSalt
  enteredWrongPassword: boolean
}

export interface PersistedRootState extends Pick<RootState, 'theme' | 'wallet' | 'network'> {}
export interface SyncedRootState extends Pick<RootState, 'theme' | 'wallet' | 'network' | 'persist'> {}
