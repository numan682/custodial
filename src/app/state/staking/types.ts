/* --- STATE --- */

import { NetworkType } from 'app/state/network/types'
import { ErrorPayload } from 'types/errors'
import { StringifiedBigInt } from 'types/StringifiedBigInt'

export interface Validator {
  address: string
  name?: string
  nodeAddress: string
  escrow: StringifiedBigInt
  status: 'active' | 'inactive' | 'unknown'
  rank: number
  media?: ValidatorMediaInfo
  current_rate?: number
  current_commission_bound?: CommissionBound
}

export interface Validators {
  timestamp: number
  network: NetworkType
  list: Validator[]
}

export interface CommissionBound {
  lower: number
  upper: number
  epochStart: number
  epochEnd?: number
}

export interface ValidatorDetails {
  currentCommissionBounds?: CommissionBound
  scheduledCommissionBounds?: CommissionBound[]
  nextCommission?: {
    fee: number
    epochStart: number
  }
}
export interface ValidatorMediaInfo {
  website_link?: string
  email_address?: string
  twitter_acc?: string
  tg_chat?: string
  logotype?: string
}

export interface Delegation {
  validator?: Validator
  amount: StringifiedBigInt
  shares: StringifiedBigInt
  validatorAddress: string
}

export interface DebondingDelegation extends Delegation {
  epoch: number
}

export interface StakingState {
  /** List of all the validators */
  validators: Validators | null

  /** Error from last attempt to update our list of validators */
  updateValidatorsError?: ErrorPayload

  /** List of active delegations for the selected account */
  delegations: Delegation[] | null

  /** List of debonding delegations for the selected account */
  debondingDelegations: DebondingDelegation[] | null

  /** Error from fetching delegations */
  updateDelegationsError?: ErrorPayload

  /** Address of the selected validator */
  selectedValidator?: string | null

  /** Lazy-loaded validator details (commission bounds and rates schedules) */
  selectedValidatorDetails: ValidatorDetails | null

  loading: boolean
}
