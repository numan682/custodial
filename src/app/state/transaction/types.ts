import { ErrorPayload } from 'types/errors'
import { StringifiedBigInt } from 'types/StringifiedBigInt'

export enum TransactionType {
  StakingTransfer = 'staking.Transfer',
  StakingAddEscrow = 'staking.AddEscrow',
  StakingReclaimEscrow = 'staking.ReclaimEscrow',
  StakingAmendCommissionSchedule = 'staking.AmendCommissionSchedule',
  StakingAllow = 'staking.Allow',
  StakingWithdraw = 'staking.Withdraw',
  RoothashExecutorCommit = 'roothash.ExecutorCommit',
  RoothashExecutorProposerTimeout = 'roothash.ExecutorProposerTimeout',
  RegistryDeregisterEntity = 'registry.DeregisterEntity',
  RegistryRegisterEntity = 'registry.RegisterEntity',
  RegistryRegisterNode = 'registry.RegisterNode',
  RegistryRegisterRuntime = 'registry.RegisterRuntime',
  GovernanceCastVote = 'governance.CastVote',
  GovernanceSubmitProposal = 'governance.SubmitProposal',
  BeaconPvssCommit = 'beacon.PVSSCommit',
  BeaconPvssReveal = 'beacon.PVSSReveal',
  BeaconVrfProve = 'beacon.VRFProve',
  ConsensusDeposit = 'consensus.Deposit',
  ConsensusWithdraw = 'consensus.Withdraw',
  ConsensusAccountsParameters = 'consensus.Parameters',
  ConsensusBalance = 'consensus.Balance',
  ConsensusAccount = 'consensus.Account',
}

export interface Transaction {
  amount: StringifiedBigInt | undefined
  fee: StringifiedBigInt | undefined
  from: string | undefined
  hash: string
  level: number | undefined
  status: boolean | undefined
  timestamp: number | undefined
  to: string | undefined
  type: TransactionType
  // These are undefined on consensus transaction
  // Only appear on ParaTime transaction
  runtimeName: string | undefined
  runtimeId: string | undefined
  round: number | undefined
}

/* --- STATE --- */
export interface TransactionState {
  step?: TransactionStep
  active: boolean
  success: boolean
  error?: ErrorPayload
  preview?: TransactionPreview
}

export type NewTransactionType = 'transfer' | 'addEscrow' | 'reclaimEscrow'
export type TransactionPayload = TransferPayload | AddEscrowPayload | ReclaimEscrowPayload

export interface TransactionPreview {
  transaction: TransactionPayload
  fee?: StringifiedBigInt
  gas?: StringifiedBigInt
}

/**
 * Flow for a transaction to be sent.
 */
export enum TransactionStep {
  Building = 'building',
  Preview = 'preview',
  Signing = 'signing',
  Submitting = 'submitting',
  Sent = 'sent',
}

export interface TransferPayload {
  type: 'transfer'

  /* bech32 Address */
  to: string

  /* Token amount */
  amount: StringifiedBigInt
}

export interface AddEscrowPayload {
  type: 'addEscrow'

  /* bech32 Address */
  validator: string

  /* Token amount */
  amount: StringifiedBigInt
}

export interface ReclaimEscrowPayload {
  type: 'reclaimEscrow'

  /* bech32 Address */
  validator: string

  /* Shares to be reclaimed */
  shares: StringifiedBigInt

  /* Displayed token equivalent */
  amount: StringifiedBigInt
}
