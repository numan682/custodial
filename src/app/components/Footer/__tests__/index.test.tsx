import { render, screen } from '@testing-library/react'
import { ResponsiveContext } from 'grommet/es6/contexts/ResponsiveContext'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { ThemeProvider } from 'styles/theme/ThemeProvider'
import { configureAppStore } from 'store/configureStore'
import { Wallet } from 'app/state/wallet/types'

import { Footer } from '..'

jest.unmock('react-i18next')

const renderComponent = (store: any, responsiveContext: string) =>
  render(
    <Provider store={store}>
      <MemoryRouter>
        <ThemeProvider>
          <ResponsiveContext.Provider value={responsiveContext}>
            <Footer />
          </ResponsiveContext.Provider>
        </ThemeProvider>
      </MemoryRouter>
    </Provider>,
  )

describe('<Footer />', () => {
  let store: ReturnType<typeof configureAppStore>

  beforeEach(() => {
    jest.resetModules()

    store = configureAppStore({
      wallet: {
        selectedWallet: 'dummy',
        wallets: {
          dummy: {
            address: 'dummy',
          } as Wallet,
        },
      },
    })
  })

  it('should render a link with commit sha', () => {
    renderComponent(store, 'large')

    expect(screen.getByRole('link', { name: 'sha0000' })).toHaveAttribute(
      'href',
      'https://github.com/oasisprotocol/oasis-wallet-web/commit/sha0000000000000000000000000000000000000',
    )
  })

  it('should render a link with version number', () => {
    renderComponent(store, 'large')

    expect(screen.getByRole('link', { name: '1.0.0-dev.1' })).toHaveAttribute(
      'href',
      'https://github.com/oasisprotocol/oasis-wallet-web/releases/tag/v1.0.0-dev.1',
    )
  })

  it('should render backend label', () => {
    renderComponent(store, 'large')

    expect(screen.getByText(/Powered by Oasis Scan API.*/)).toBeInTheDocument()
  })

  it('should render mobile version of footer', () => {
    renderComponent(store, 'small')

    expect(screen.getByRole('contentinfo')).toMatchSnapshot()
  })
})
