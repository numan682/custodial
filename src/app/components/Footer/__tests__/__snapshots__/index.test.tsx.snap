// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render mobile version of footer 1`] = `
.c8 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #666666;
  stroke: #666666;
}

.c8 g {
  fill: inherit;
  stroke: inherit;
}

.c8 *:not([stroke])[fill='none'] {
  stroke-width: 0;
}

.c8 *[stroke*='#'],.c8 *[STROKE*='#'] {
  stroke: inherit;
  fill: none;
}

.c8 *[fill-rule],
.c8 *[FILL-RULE],
.c8 *[fill*='#'],.c8 *[FILL*='#'] {
  fill: inherit;
  stroke: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-left: 24px;
  padding-right: 24px;
  padding-top: 1rem;
  padding-bottom: 4rem;
  border-radius: 5px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-left: 24px;
  padding-right: 24px;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin: 6px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  box-sizing: border-box;
  font-size: 14px;
  line-height: 20px;
  color: #0092f6;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c2:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c1 {
  margin-bottom: 12px;
  font-size: 14px;
  line-height: 20px;
  text-align: center;
}

.c3 {
  font-size: 14px;
  line-height: 20px;
  text-align: center;
}

.c9 {
  font-size: 14px;
  line-height: 20px;
}

.c5 {
  background-color: #FFFFFF;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  height: 4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  box-shadow: 0px 0px 1px #EDEDED;
  border-top: solid 1px #11111108;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

@media only screen and (max-width:768px) {
  .c0 {
    margin-bottom: 24px;
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    padding-left: 12px;
    padding-right: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    padding-top: 1rem;
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    padding-bottom: 4rem;
  }
}

@media only screen and (max-width:768px) {
  .c6 {
    padding-left: 12px;
    padding-right: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c7 {
    margin: 3px;
  }
}

<footer
  class="c0"
>
  <span
    class="c1"
  >
    <span>
      Oasis Wallet is fully 
    </span>
    <a
      class="c2"
      href="https://github.com/oasisprotocol/oasis-wallet-web/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span>
        open source
      </span>
    </a>
    <span>
       - Feedback and issues are appreciated!
    </span>
  </span>
  <span
    class="c1"
  >
    <a
      class="c2"
      href="https://wallet.oasis.io/t-c"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span>
        Terms and Conditions
      </span>
    </a>
  </span>
  <span
    class="c3"
  >
    <span>
      Version: 
    </span>
    <a
      class="c2"
      href="https://github.com/oasisprotocol/oasis-wallet-web/releases/tag/v1.0.0-dev.1"
      rel="noopener noreferrer"
      target="_blank"
    >
      1.0.0-dev.1
    </a>
    <span>
       (commit: 
    </span>
    <a
      class="c2"
      href="https://github.com/oasisprotocol/oasis-wallet-web/commit/sha0000000000000000000000000000000000000"
      rel="noopener noreferrer"
      target="_blank"
    >
      sha0000
    </a>
    <span>
      ) built at 1/1/1970, 12:00:00 AM
    </span>
    <div
      class="c4"
    >
      Powered by Oasis Scan API & Oasis gRPC
    </div>
  </span>
  <nav
    class="c5"
    data-testid="mobile-footer-navigation"
  >
    <a
      class=""
      href="/account/dummy"
    >
      <span
        class="c6"
      >
        <span
          class="c7"
        >
          <svg
            aria-label="Money"
            class="c8"
            viewBox="0 0 24 24"
          >
            <path
              d="M16 16c0-1.105-3.134-2-7-2s-7 .895-7 2 3.134 2 7 2 7-.895 7-2zM2 16v4.937C2 22.077 5.134 23 9 23s7-.924 7-2.063V16M9 5c-4.418 0-8 .895-8 2s3.582 2 8 2M1 7v5c0 1.013 3.582 2 8 2M23 4c0-1.105-3.1-2-6.923-2-3.824 0-6.923.895-6.923 2s3.1 2 6.923 2S23 5.105 23 4zm-7 12c3.824 0 7-.987 7-2V4M9.154 4v10.166M9 9c0 1.013 3.253 2 7.077 2C19.9 11 23 10.013 23 9"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </span>
        <span
          class="c9"
        >
          Wallet
        </span>
      </span>
    </a>
    <a
      class=""
      href="/account/dummy/stake"
    >
      <span
        class="c6"
      >
        <span
          class="c7"
        >
          <svg
            aria-label="LineChart"
            class="c8"
            viewBox="0 0 24 24"
          >
            <path
              d="m1 16 7-7 5 5L23 4M0 22h23.999M16 4h7v7"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </span>
        <span
          class="c9"
        >
          Stake
        </span>
      </span>
    </a>
    <a
      class=""
      href="/account/dummy/paratimes"
    >
      <span
        class="c6"
      >
        <span
          class="c7"
        >
          <svg
            aria-label="Inherit"
            class="c8"
            viewBox="0 0 24 24"
          >
            <path
              d="m17 18-5-3 5 3zM7 18l5-3v-4m5 9a3 3 0 1 0 6 0 3 3 0 0 0-6 0zM4 17a3 3 0 1 0 0 6 3 3 0 0 0 0-6zM17 6a5 5 0 1 1-10.001-.001A5 5 0 0 1 17 6z"
              fill="none"
              stroke="#231F20"
              stroke-width="2"
            />
          </svg>
        </span>
        <span
          class="c9"
        >
          ParaTimes
        </span>
      </span>
    </a>
    <a
      class=""
      href="/account/dummy/fiat"
    >
      <span
        class="c6"
      >
        <span
          class="c7"
        >
          <svg
            aria-label="CreditCard"
            class="c8"
            viewBox="0 0 24 24"
          >
            <path
              d="M1 5c0-.552.44-1 1.002-1h19.996A1 1 0 0 1 23 5v14c0 .552-.44 1-1.002 1H2.002A1 1 0 0 1 1 19V5zm0 3h22v2H1V8zm4 7h2v.5H5V15zm5 0h6v.5h-6V15z"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </span>
        <span
          class="c9"
        >
          Buy
        </span>
      </span>
    </a>
  </nav>
</footer>
`;
